content/posts/unpublished/styling-your-legacy-micro-frontends/post.md

```diff
@@ -0,0 +1,135 @@
+* Combine two themes into one, leading to duplication of styles
+
+Each option has its own advantages and drawbacks, so it's important to delve deeper into the details.
+
+## One theme
+
+This method requires more effort to maintain as there is no straightforward approach. You'll have to visually scan the app and fix styling problems one by one, including new CSS code as you go.
+
+Some of these problems are harder to solve as you have to deal with overriding the styles of the shell app, and the final look might not be pixel perfect.
+
+Still, the reward for this effort is a smaller bundle while you keep mastering CSS.
+
+### Bootstrap changes from 4 to 5
+
+One good place to start looking is the official Bootstrap [migration](https://getbootstrap.com/docs/5.0/migration/) changelog from 4 to 5, but don't worry, because not necessarily all the changes apply to your situation.
+
+In the Backbase scenario, the renaming of the "screen reader" classes to "visually hidden" hits hard as you start seeing things that should not be on the screen, but at the same time is quite easy to solve.
+
+### Set-up
+
+1. Start by isolating the changes in your remote `AppComponent`.
+```angular2html
+<div class="angular17-theme">
+  <router-outlet></router-outlet>
+</div>
+```
+
+2. In the `styles.scss` of your shell app, use the same class you added to the remote `AppComponent`.
+
+```scss
+@import '@backbase/backbase-theme-business-preset/scss/main';
+
+.angular17-theme {}
+```
+
+3. Now you can start fixing issues; for example, for the mentioned new "visually hidden" classes, you can just extend the old "screen reader" ones.
+```scss
+.angular17-theme {
+  .visually-hidden {
+    @extend .sr-only;
+  }
+}
+```
+
+## Two themes
+
+Setting up this approach can be quite challenging as it requires including both Bootstrap versions and, in the Backbase scenario, incorporating both versions of the UI-Ang library along with having two themes.
+
+> **_NOTE:_**   Backbase [design system](https://designsystem.backbase.com/) is a collection of guidelines, tools, and resources to help build superior banking experiences. The `@backbase/ui-ang` library exposes the UI components for web and since the version 9 it depends on Bootstrap 5.
+
+Including coupled packages results in a significant amount of code duplication, but once it's set and configured you can nearly forget about making further changes to your CSS.
+
+### Set-up
+
+1. Create a `/themes` folder at the root level.
+2. Go to the newly created folder and install UI-Ang with its dependencies.
+```npm
+npm i @backbase/ui-ang@10 ngx-quill@16
+```
+3. Copy `/backbase-theme-preset-template` from the `/node_modules/@backbase/ui-ang/dist` to the `/themes` folder.
+4. Update the `/shell/src/styles.scss` to include the new theme.
+```scss
+@import '@backbase/backbase-theme-business-preset/scss/main';
+
+.angular17-theme {
+  @import '../../themes/backbase-theme-preset-template/scss/main';
```

Comment:
Can we avoid using the name `backbase-theme-preset-template` here? These templates named like this are deprecated, so I'd rather call it as `angular-17-theme` or `remote-theme` to avoid misunderstanding